src/main/java/nextstep/courses/service/SessionService.java

```diff
@@ -0,0 +1,35 @@
+package nextstep.courses.service;
+
+import nextstep.courses.domain.Session;
+import nextstep.courses.domain.SessionRepository;
+import nextstep.users.domain.NsUser;
+import nextstep.users.domain.UserRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@Transactional
+public class SessionService {
+    private final SessionRepository sessionRepository;
+
+    public SessionService(SessionRepository sessionRepository) {
+        this.sessionRepository = sessionRepository;
+    }
+
+    public long save(Session session) {
+        return sessionRepository.save(session);
+    }
+
+    @Transactional(readOnly = true)
+    public Session findById(long id) {
+        return sessionRepository.findById(id);
+    }
+
+    public long register(Session session, NsUser nsUser) {
+        session.register(nsUser);
+        return sessionRepository.saveSessionUser(session, nsUser);
+    }
```

Comment:
```suggestion
    public long register(Long sessionId, NsUser nsUser) {
        Session session = sessionRepository.findById(sessionId);
        Student student = session.register(nsUser);
        sessionRepository.saveSessionUser(student);
    }
```

보통 Session 객체가 외부에서 전달되기 보다 id를 전달한 후 조회하는 방식이지 않을까?
`sessionRepository.saveSessionUser(session, nsUser)`와 같이 구현하는 것도 좋지만 SessionUser(앞 예에서는 Student)로 값을 전달하는 것이 추후에 요구사항이 변경되더라도 변경을 최소화할 수 있지 않을까?